### PR TITLE
onionshare: 2.6 -> 2.6.1

### PR DESCRIPTION
**Pull request by aktaboot**
## Description of changes

- Update onionshare
- add new dependencies 
- remove no longer needed patch

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@bbjubjub2494 

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/onionshare/default.nix [hunk 1]
```diff
@@ -91,6 +95,9 @@ rec {
       pynacl
       requests
       unidecode
+      waitress
+      python3Packages.qrcode
+      python3Packages.pyside6
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/onionshare/default.nix:100]:
```suggestion
      pynacl
      pyside6
      qrcode
      requests
      unidecode
      waitress
```

File: pkgs/applications/networking/onionshare/default.nix [hunk 2]
```diff
@@ -25,15 +25,18 @@
 , tor
 , obfs4
 , snowflake
+, waitress
+, flask-compress
+, python3Packages
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/onionshare/default.nix:30]:
```suggestion
```
please add the individual packages instead and try to keep the alphabetical sorting.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by bbjubjub2494 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
I've pushed a change to ignore the failing test on darwin and sort all the listed packages.

**Comment by happysalada** [comment]
@SuperSandro2000 this looks good to me to merge ? Would you mind approving the PR ? Or are there other things that need changing ?

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**